client/src/components/App.jsx

```diff
@@ -1,77 +1,128 @@
 import React from 'react';
-// import axios from 'axios';
 import LineChartContainer from './LineChartContainer';
 import TimeFilter from './TimeFilter';
 import StockInfo from './StockInfo';
 import CompanyInfo from './CompanyInfo';
 import TagContainer from './TagContainer';
-import { BrowserRouter as Router, Route, Link } from "react-router-dom";
 import API from './api';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      chartData: null,
-      stockInfo: null,
+      stockId: null,
+      chartData: {},
       averageStock: null,
       changePercent: null,
       selectedFilter: 'day',
-      currentPrice: null
+      currentPrice: null,
+      stockCompany: null,
+      relatedTags: [],
+      noOfOwners: null,
+      recommendationPercent: null,
     };
   }
   
 
   componentDidMount() {
     const { stockId } = this.props.match ? this.props.match.params : { stockId: null };
-    API.get((stockId && `/api/${stockId}`) || `/api/TSLA`)
+    API.get((stockId && `/api/stocks/${stockId}`) || `/api/stocks/TSLA`)
```

Comment:
Nice, the api [library](https://www.npmjs.com/package/api) does simplify the syntax of an http get request